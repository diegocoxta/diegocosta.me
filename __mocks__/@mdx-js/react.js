module.exports = {
  MDXProvider: jest.fn(({ children }) => children),
};
