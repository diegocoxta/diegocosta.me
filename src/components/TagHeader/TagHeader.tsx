import React from 'react';
import styled from 'styled-components';

import Container from '~/components/Container';

const Content = styled.h1`
  font-size: 28px;
  background: ${({ theme }) => theme.accentColor};
  display: inline-block;
  padding: 5px;
  margin: 40px 0 20px 0;
`;

interface TagHeaderProps {
  name: string;
  count: number;
}

export default function TagHeader(props: TagHeaderProps): React.ReactElement {
  return (
    <Container>
      <Content>
        {props.name} ({props.count})
      </Content>
    </Container>
  );
}
