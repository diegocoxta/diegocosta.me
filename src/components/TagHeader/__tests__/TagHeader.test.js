import React from 'react';
import { render } from '@testing-library/react';

import TagHeader from '../TagHeader';

describe('<TagHeader />', () => {
  it('renders properly', () => {
    const { baseElement, queryByText } = render(<TagHeader name="John Doe" count={0} />);
    expect(queryByText('John Doe (0)')).toBeTruthy();
    expect(baseElement).toMatchSnapshot();
  });
});
