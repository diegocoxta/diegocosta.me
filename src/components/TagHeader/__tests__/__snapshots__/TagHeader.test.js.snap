// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TagHeader /> should render properly 1`] = `
.c0 {
  max-width: 800px;
  margin: 0 auto;
  padding: 0 20px;
}

.c1 {
  font-size: 28px;
  display: inline-block;
  padding: 5px;
  margin: 40px 0 20px 0;
}

<body>
  <div>
    <section
      class="c0"
    >
      <main>
        <h1
          class="c1"
        >
          John Doe
           (
          0
          )
        </h1>
      </main>
    </section>
  </div>
</body>
`;
